addons/survey/views/survey_templates.xml

```diff
@@ -186,8 +187,11 @@
             t-att-data-time-limit-minutes="time_limit_minutes"/>
         <t t-if="survey.questions_layout == 'one_page'">
             <t t-foreach='survey.question_and_page_ids' t-as='question'>
-                <h2 t-if="question.is_page" t-field='question.title' class="o_survey_title pb16 text-break w-lg-50 mx-lg-auto" />
-                <div t-if="question.is_page" t-field='question.description' class="text-break w-lg-50 mx-lg-auto"/>
+                <t t-if="question.is_page" t-set="display_section" t-value="all(question_id.triggering_answer_ids for question_id in question.question_ids)"/>
```

Comment:
I'm not sure about the initial condition: We display the section if all questions have a triggering answer ? It does not seem to check if all questions of the section are hidden :thinking: There is maybe a missing condition or something I didn't understand...

Reply:
Yes you're right I changed it to:
```xml
<t t-set="display_section" t-value="any(not q.triggering_answer_ids for q in page.question_ids)
      or (page.question_ids and survey.questions_selection == 'random')"/>
```
Meaning that it will display the section if there is at least one non-conditionally triggered question in it, or when the selection is random and there's at least one question in the section.
For ref, the questions visibility condition here: https://github.com/odoo/odoo/blob/20d7f6e26c8228d0cb2010b1021bb367b697260a/addons/survey/views/survey_templates.xml#L313-L318